packages/grid/x-data-grid/src/components/columnHeaders/GridColumnGroupHeader.tsx

```diff
@@ -46,8 +43,8 @@ const useUtilityClasses = (ownerState: OwnerState) => {
       headerAlign === 'center' && 'columnHeader--alignCenter',
       headerAlign === 'right' && 'columnHeader--alignRight',
       isDragging && 'columnHeader--moving',
-      showRightBorder && 'columnHeader--withRightBorder',
       showColumnBorder && 'columnHeader--showColumnBorder',
```

Comment:
I didn't combine these two as they apparently serve different purposes and might be segregated again in future